src/nvm.go

```
@@ -514,17 +518,13 @@ func findLatestSubVersion(version string, localOnly ...bool) string {
 }
 
 func use(version string, cpuarch string, reload ...bool) {
-	v, a, err := getVersion(version, cpuarch, true)
-	version = v
-	cpuarch = a
+	version, cpuarch, err := getVersion(version, cpuarch, true)
```

@vladonemo is this the line that's preventing the main use case from succeeding?

---

the call returns an installed version, but it also returns the error in case user only provided part of the SemVer.

---

The code should check to see if the error message indicates it is a partial version. If it _is_ a partial, the error can be ignored. If it is **not** a partial, the error should be thrown.

---

right. And this is what the code does at the moment - line 524. If the error is **not** a partial, it fails with this error. Otherwise it continues, because the `getVersion` could resolve the latest sub version.

---

apologies - I was just looking at the changes, not the full file. I see this is the logic that exists.